src/components/TextField/TextField.tsx

```diff
@@ -1,18 +1,15 @@
-import React from 'react';
+import React, {useState, useEffect, useRef, useCallback} from 'react';
 import {addEventListener} from '@shopify/javascript-utilities/events';
-import {createUniqueIDFactory} from '@shopify/javascript-utilities/other';
 import {CircleCancelMinor} from '@shopify/polaris-icons';
 import {VisuallyHidden} from '../VisuallyHidden';
 import {classNames, variationName} from '../../utilities/css';
-
+import {useI18n} from '../../utilities/i18n';
+import {useMountedRef} from '../../utilities/use-mounted-ref';
```

Comment:
I brought this over from `@shopify/react-hooks`. I'm thinking it might be useful to add?